tools/RAiDER/statsPlot.py

```diff
@@ -527,6 +532,21 @@ def _convert_SI(self, val, unit_in, unit_out):
 
         return val*SI[unit_in]/SI[unit_out]
 
+    def _check_stationgrid_intersection(self, stat_ID):
+        '''
+        Return index of grid cell which intersects with station
+        Note: Fast, but assumes station locations don't change
+        '''
+        coord = Point((self.unique_points[1][self.unique_points[0].index(
+            stat_ID)], self.unique_points[2][self.unique_points[0].index(stat_ID)]))
+        # Get grid cell polygon which intersect with station coordinate
+        grid_int = self.polygon_tree.query(coord)[0]
+        # Pass corresponding grid cell index
+        if grid_int == []:
+            return stat_ID, 'NaN'
+        else:
+            return stat_ID, self.polygon_dict[id(grid_int)]
```

Comment:
I would write it like this:
```suggestion
        if grid_int:
            return stat_ID, self.polygon_dict[id(grid_int)]
        return stat_ID, 'NaN'
```

Also I don't think I've ever seen `id` actually used in python before.

Reply:
Yeah, I think it's a clean way of handling it in this case instead of passing a large dictionary filled with polygons spanning the global. Also a new idea I encountered in the depths of stackoverflow!